Client/Frontend/Browser/Onboarding/OnboardingAdsFinishedView.swift

```diff
@@ -0,0 +1,110 @@
+        static let descriptionContentInset: CGFloat = 32
+        static let animationContentInset: CGFloat = 50.0
+    }
+    
+    class View: UIView {
+        
+        let finishButton = CommonViews.primaryButton(text: Strings.OBFinishButton).then {
+            $0.accessibilityIdentifier = "OnboardingAdsFinishedViewController.StartBrowsing"
+        }
+        
+        private let mainStackView = UIStackView().then {
+            $0.axis = .vertical
+            $0.spacing = UX.negativeSpacing
+        }
+        
+        let imageView = AnimationView(name: "onboarding-ads").then {
+            $0.contentMode = .scaleAspectFit
+            $0.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+            $0.backgroundBehavior = .pauseAndRestore
+            $0.loopMode = .playOnce
+            $0.currentProgress = 1.0
+        }
+        
+        private let descriptionView = UIView().then {
+            $0.layer.cornerRadius = 12
+            $0.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+        }
+        
+        private let descriptionStackView = UIStackView().then {
+            $0.axis = .vertical
+            $0.spacing = 32
+        }
+        
+        private let textStackView = UIStackView().then { stackView in
+            stackView.axis = .horizontal
+            stackView.spacing = 8
+            
+            let titleLabel = CommonViews.secondaryText(Strings.OBAdsInfoDescription)
+            [UIView.spacer(.horizontal, amount: 20),
+             titleLabel,
+             UIView.spacer(.horizontal, amount: 20)]
+                .forEach {
+                    stackView.addArrangedSubview($0)
+                }
+        }
+        
+        private let buttonsStackView = UIStackView().then {
+            $0.distribution = .equalCentering
+        }
+        
+        init(theme: Theme, themeColour: UIColor) {
+            super.init(frame: .zero)
+            
+            descriptionView.backgroundColor = themeColour
+            
+            mainStackView.tag = OnboardingViewAnimationID.details.rawValue
+            descriptionStackView.tag = OnboardingViewAnimationID.detailsContent.rawValue
+            imageView.tag = OnboardingViewAnimationID.background.rawValue
+            
+            addSubview(imageView)
+            addSubview(mainStackView)
+            mainStackView.snp.makeConstraints {
+                $0.leading.equalTo(self.safeArea.leading)
+                $0.trailing.equalTo(self.safeArea.trailing)
+                $0.bottom.equalTo(self.safeArea.bottom)
+            }
+            
+            descriptionView.addSubview(descriptionStackView)
+            descriptionStackView.snp.makeConstraints {
+                $0.edges.equalToSuperview().inset(UX.descriptionContentInset)
+            }
+            
+            [descriptionView].forEach(mainStackView.addArrangedSubview(_:))
+
+            [UIView.spacer(.horizontal, amount: 0),
```

Comment:
This doesn't work. You can't set custom spacing "before". It also depends on the layout and since this layout requires "centering", a blank view with lower hugging before allows it to achieve equal centering and proper spacing. Without it, the UI is completely broken.